pkg/engine/runtime/init/init.go

```diff
@@ -1,18 +1,48 @@
 package init
 
 import (
+	"fmt"
+	"reflect"
+
 	"kusionstack.io/kusion/pkg/engine/models"
 	"kusionstack.io/kusion/pkg/engine/runtime"
 	"kusionstack.io/kusion/pkg/engine/runtime/terraform"
+	"kusionstack.io/kusion/pkg/status"
 )
 
-func InitRuntime() map[models.Type]InitFn {
-	runtimes := map[models.Type]InitFn{
-		runtime.Kubernetes: runtime.NewKubernetesRuntime,
-		runtime.Terraform:  terraform.NewTerraformRuntime,
-	}
-	return runtimes
+var SupportRuntimes = map[models.Type]InitFn{
```

Comment:
map to func will execute NewXXRuntime() for each resource node, can we just run NewXXRuntime only once for each apply action.

Reply:
In resource_node.go, every resource node need to get live state from runtime, and apply or destroy it.
And `RuntimeMap[resourceType]` returns func `NewXXRuntime()`, not a singleton variable, so Runtime will be initialized many times, not only in one operation, but also in one resource node.

Reply:
In resource_node.go, every resource node need to get live state from runtime, and apply or destroy it.
And RuntimeMap[resourceType] returns func NewXXRuntime(), not a singleton variable, so Runtime will be initialized many times, not only in one operation, but also in one resource node.

Reply:
RuntimeMap  is `map[models. Type]runtime. Runtime` not `map[models. Type]InitFn`

Reply:
WHAT? are you sure?
```go
var SupportRuntimes = map[models.Type]InitFn{
	runtime.Kubernetes: runtime.NewKubernetesRuntime,
	runtime.Terraform:  terraform.NewTerraformRuntime,
}

// InitFn runtime init func
type InitFn func() (runtime.Runtime, error)
```

Reply:
We can make a issue to talk about this